src/clr-angular/forms/common/if-error.spec.ts

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+import {Component} from "@angular/core";
+import {TestBed} from "@angular/core/testing";
+import {FormsModule} from "@angular/forms";
+import {By} from "@angular/platform-browser";
+
+import {ClrIconModule} from "../../icon/icon.module";
+import {ClrInput} from "../input/input";
+import {ClrInputContainer} from "../input/input-container";
+
+import {ClrControlError} from "./error";
+import {ClrIfError} from "./if-error";
+
+const errorMessage = "ERROR_MESSAGE";
+
+@Component({template: `<div *clrIfError></div>`})
+class InvalidUseTest {
+}
+
+@Component({
+    template: `
+    <clr-input-container>
+        <input type="text" name="model" clrInput [(ngModel)]="model" required />
```

Comment:
As mentioned before, these tests use too many components at the same time, they're not isolated at all. I'd much rather you just declared the `ControlService` as a provider and that you instantiated a `new FormControl()` for this spec. 
FormControls are quite simple to instantiate, and they offer methods like `setErrors()` for you to have full control of them to test your directive.